packages/fxa-auth-server/test/scripts/stripe-products-and-plans-converter.js

```diff
@@ -0,0 +1,798 @@
+        id: 'plan_123',
+      });
+      const expectedPlanConfig = {
+        active: true,
+        stripePriceId: testPlan.id,
+        capabilities: {
+          aFakeClientId12345: ['more', 'comma', 'separated', 'values'],
+        },
+        uiContent: {
+          upgradeCTA: 'hello <a href="http://example.org">world</a>',
+        },
+        urls: {
+          download: 'https://example.com/download',
+        },
+        productOrder: 2,
+      };
+      const actualPlanConfig = converter.stripePlanToPlanConfig(testPlan);
+      assert.deepEqual(expectedPlanConfig, actualPlanConfig);
+      const { error } = await PlanConfig.validate(actualPlanConfig);
+      assert.isUndefined(error);
+    });
+  });
+
+  describe('findLocaleStringFromStripePlan', () => {
+    it('returns a locale from the plan nickname if valid', () => {
+      const planWithLocalizedData = {
+        ...deepCopy(plan),
+        nickname: '123Done Pro Monthly es-ES',
+        metadata: {
+          'product:details:1': 'Producto nuevo',
+        },
+      };
+      const expected = 'es-ES';
+      const actual = converter.findLocaleStringFromStripePlan(
+        planWithLocalizedData
+      );
+      assert.equal(expected, actual);
+    });
+    it('returns null if no valid locale is found', () => {
+      const planWithLocalizedData = {
+        ...deepCopy(plan),
+        nickname: '123Done Pro Monthly',
+        metadata: {
+          'product:details:1': 'Producto nuevo',
+        },
+      };
+      const actual = converter.findLocaleStringFromStripePlan(
+        planWithLocalizedData
+      );
+      assert.isNull(actual);
+    });
+  });
+
+  describe('stripePlanLocalesToProductConfigLocales', () => {
+    it('returns a ProductConfig.locales object if a locale is found', () => {
+      const planWithLocalizedData = {
+        ...deepCopy(plan),
+        nickname: '123Done Pro Monthly es-ES',
+        metadata: {
+          'product:details:1': 'Producto nuevo',
+          'product:details:2': 'Mas mejor que el otro',
+        },
+      };
+      const expected = {
+        'es-ES': {
+          uiContent: {
+            details: ['Producto nuevo', 'Mas mejor que el otro'],
+          },
+          urls: {},
+          support: {},
+        },
+      };
+      const actual = converter.stripePlanLocalesToProductConfigLocales(
+        planWithLocalizedData
+      );
+      assert.deepEqual(expected, actual);
+    });
+    it('returns {} if no locale is found', () => {
+      const planWithLocalizedData = {
+        ...deepCopy(plan),
+        nickname: '123Done Pro Monthly',
+        metadata: {
+          'product:details:1': 'Producto nuevo',
+          'product:details:2': 'Mas mejor que el otro',
+        },
+      };
+      const expected = {};
+      const actual = converter.stripePlanLocalesToProductConfigLocales(
+        planWithLocalizedData
+      );
+      assert.deepEqual(expected, actual);
+    });
+  });
+
+  describe('convert', async () => {
```

Comment:
The tests in this block are sometimes timing out (~10% of the time) even increasing the timeout to 10s. There must be some race condition that isn't addressed. Any hypotheses? It should work similarly to the `test/local/payments/configuration/manager.js` test.

Reply:
Paired with Ben to resolve the test flakiness -- see https://github.com/mozilla/fxa/pull/12040/commits/7d69607096add0fb81eda26385f20031505eb631.